libraries/plugins/custom_operations/custom_evaluators.cpp

```diff
@@ -0,0 +1,132 @@
+   vector<object_id_type> results;
+
+   if (op.extensions.value.remove.valid() && *op.extensions.value.remove)
+   {
+      for(auto const& row: *op.extensions.value.key_values) {
+         auto itr = index.find(make_tuple(_account, *op.extensions.value.catalog, row.first));
+         if(itr != index.end()) {
+            results.push_back(itr->id);
+            _db->remove(*itr);
+         }
+      }
+   }
+   else {
+      for(auto const& row: *op.extensions.value.key_values) {
+         if(row.first.length() > CUSTOM_OPERATIONS_MAX_KEY_SIZE)
+         {
+            wlog("Key can't be bigger than ${max} characters", ("max", CUSTOM_OPERATIONS_MAX_KEY_SIZE));
+            continue;
+         }
+         auto itr = index.find(make_tuple(_account, *op.extensions.value.catalog, row.first));
+         if(itr == index.end())
+         {
+            try {
+               auto created = _db->create<account_storage_object>( [&op, this, &row]( account_storage_object& aso ) {
+                  aso.catalog = *op.extensions.value.catalog;
+                  aso.account = _account;
+                  aso.key = row.first;
+                  aso.value = fc::json::from_string(row.second);
+               });
+               results.push_back(created.id);
+            }
+            catch(const fc::parse_error_exception& e) { wdump((e.to_detail_string())); }
+            catch(const fc::assert_exception& e) { wdump((e.to_detail_string())); }
+         }
+         else
+         {
+            try {
+               _db->modify(*itr, [&op, this, &row](account_storage_object &aso) {
+                  aso.value = fc::json::from_string(row.second);
+               });
+               results.push_back(itr->id);
+            }
+            catch(const fc::parse_error_exception& e) { wdump((e.to_detail_string())); }
+            catch(const fc::assert_exception& e) { wdump((e.to_detail_string())); }
+         }
+      }
+   }
+   return results;
+}
+vector<object_id_type> custom_generic_evaluator::do_apply(const account_storage_list& op)
+{
+   auto &index = _db->get_index_type<account_storage_index>().indices().get<by_account_catalog_subkey>();
+   vector<object_id_type> results;
+
+   if (op.extensions.value.remove.valid() && *op.extensions.value.remove)
+   {
+      for(auto const& list_value: *op.extensions.value.values) {
+
+         auto itr = index.find(make_tuple(_account, *op.extensions.value.catalog, list_value));
+         if(itr != index.end()) {
+            results.push_back(itr->id);
+            _db->remove(*itr);
+         }
+      }
+   }
+   else {
+      for(auto const& list_value: *op.extensions.value.values) {
+         if(list_value.length() > 200)
+         {
+            wlog("List value can't be bigger than ${max} characters", ("max", CUSTOM_OPERATIONS_MAX_KEY_SIZE));
+            continue;
+         }
+         auto itr = index.find(make_tuple(_account, *op.extensions.value.catalog, list_value));
+         if(itr == index.end())
```

Comment:
A list that cannot contain duplicate values is not a list but an unordered set. Is this the desired behaviour?
A set is better implemented as a map where the values are ignored. And we have map functionality already.

Reply:
I think that is the desired functionality. I removed subkey here: https://github.com/bitshares/bitshares-core/pull/1926/commits/ff11b33193aac157a8837fd9d1e7d7555eaa875f

Do you think we should remove the list operation all together and just have the map ?

Reply:
> remove the list operation all together and just have the map ?

Hm.
An actual list could be emulated by using map keys "1", "2", "3", ...
I think that would work.